File: 251003/beskosty/lab3-5/publisher/src/main/java/com/example/lab1/mapper/IssueMapper.java
package com.example.lab1.mapper;

import com.example.lab1.dto.IssueRequestTo;
import com.example.lab1.dto.IssueResponseTo;
import com.example.lab1.model.Issue;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface IssueMapper {
    @Mapping(target = "created", expression = "java(java.time.LocalDateTime.now())")
    @Mapping(target = "modified", expression = "java(java.time.LocalDateTime.now())")
    Issue toEntity(IssueRequestTo dto);

    IssueResponseTo toDto(Issue entity);
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/mapper/IssueMapper.kt
package bsuir.dc.rest.mapper

import bsuir.dc.rest.dto.from.IssueRequestTo
import bsuir.dc.rest.dto.to.IssueResponseTo
import bsuir.dc.rest.entity.Issue
import bsuir.dc.rest.entity.Label
import bsuir.dc.rest.entity.Writer

fun IssueRequestTo.toEntity(writer: Writer, labels: MutableSet<Label>) = Issue(
    id = this.id,
    writer = writer,
    title = this.title,
    content = this.content,
    labels = labels,
)

fun Issue.toResponse() = IssueResponseTo(
    id = this.id,
    writerId = this.writer.id,
    title = this.title,
    content = this.content,
    created = this.created,
    modified = this.modified
)
